File: CloudFlow/FrontEnd-App/src/app/shared/model/permission.model.ts
import { JsonObject, JsonProperty } from "json2typescript";

@JsonObject
export class PermissionModel {

	public static readonly TYPE_USER: number = 1;
	public static readonly TYPE_ORGANIZATION_UNIT: number = 2;
	public static readonly TYPE_GROUP: number = 3;

	public static readonly PERMISSION_COORDINATOR: string = "1111";
	public static readonly PERMISSION_COLABORATOR: string = "1110";
	public static readonly PERMISSION_EDITOR: string = "1100";
	public static readonly PERMISSION_READER: string = "1000";

	@JsonProperty("entityId", String)
	public entityId: string = null;

	@JsonProperty("entityName", String, true)
	public entityName: string = null;

	@JsonProperty("entityType", Number)
	public entityType: number = null;

	@JsonProperty("permission", String)
	public permission: string = null;

	public isGroup(): boolean {
		return this.entityType === PermissionModel.TYPE_GROUP;
	}

	public isOrganizationUnit(): boolean {
		return this.entityType === PermissionModel.TYPE_ORGANIZATION_UNIT;
	}

	public isUser(): boolean {
		return this.entityType === PermissionModel.TYPE_USER;
	}
}

File: TaxiApp/src/ro/taxiApp/docs/presentation/client/shared/model/PermissionModel.java
package ro.taxiApp.docs.presentation.client.shared.model;

import com.extjs.gxt.ui.client.data.BaseModel;
import com.google.gwt.user.client.rpc.IsSerializable;

public class PermissionModel extends BaseModel implements IsSerializable {

	private static final long serialVersionUID = -5564656544137688401L;
	
    public final static int TYPE_USER = 1;
    public final static int TYPE_ORGANIZATION_UNIT = 2;
    public final static int TYPE_GROUP = 3;
    
    public final static String PERMISSION_COORDINATOR = "1111";
    public final static String PERMISSION_COLABORATOR = "1110";
    public final static String PERMISSION_EDITOR = "1100";
    public final static String PERMISSION_READER = "1000";

    public static final String PROPERTY_ENTITY_ID = "entityId";
    public static final String PROPERTY_ENTITY_NAME = "entityName";
	public static final String PROPERTY_ENTITY_TYPE = "entityType";
    public static final String PROPERTY_PERMISSION = "permission";

    public PermissionModel() {}

    public PermissionModel(int entityType, String permission, String userId, String entityName) {
        this();
        setEntityType(entityType);
        setPermission(permission);
        setEntityId(userId);
        setEntityName(entityName);
    }

    public Integer getEntityType() {
        return get(PROPERTY_ENTITY_TYPE);
    }

    public void setEntityType(Integer entityType) {
        set(PROPERTY_ENTITY_TYPE, entityType);
    }

    public String getPermission() {
        return get(PROPERTY_PERMISSION);
    }

    public void setPermission(String permission) {
        set(PROPERTY_PERMISSION, permission);
    }

    public String getEntityId() {
        return get(PROPERTY_ENTITY_ID);
    }

    public void setEntityId(String entityId) {
        set(PROPERTY_ENTITY_ID, entityId);
    }

    public String getEntityName() {
        return (String) get(PROPERTY_ENTITY_NAME);
    }

    public void setEntityName(String entityName) {
        set(PROPERTY_ENTITY_NAME, entityName);
    }
}